crates/bevy_render/src/render_asset.rs

```diff
@@ -19,27 +21,41 @@ pub enum PrepareAssetError<E: Send + Sync + 'static> {
 ///
 /// In the [`ExtractSchedule`] step the asset is transferred
 /// from the "main world" into the "render world".
-/// Therefore it is converted into a [`RenderAsset::ExtractedAsset`], which may be the same type
-/// as the render asset itself.
 ///
 /// After that in the [`RenderSet::PrepareAssets`] step the extracted asset
 /// is transformed into its GPU-representation of type [`RenderAsset::PreparedAsset`].
-pub trait RenderAsset: Asset {
-    /// The representation of the asset in the "render world".
-    type ExtractedAsset: Send + Sync + 'static;
+pub trait RenderAsset: Asset + Clone {
     /// The GPU-representation of the asset.
     type PreparedAsset: Send + Sync + 'static;
+
     /// Specifies all ECS data required by [`RenderAsset::prepare_asset`].
+    ///
     /// For convenience use the [`lifetimeless`](bevy_ecs::system::lifetimeless) [`SystemParam`].
     type Param: SystemParam;
-    /// Converts the asset into a [`RenderAsset::ExtractedAsset`].
-    fn extract_asset(&self) -> Self::ExtractedAsset;
-    /// Prepares the `extracted asset` for the GPU by transforming it into
-    /// a [`RenderAsset::PreparedAsset`]. Therefore ECS data may be accessed via the `param`.
+
+    /// Whether or not to unload the asset after extracting it to the render world.
+    fn unload_after_extract(&self) -> RenderAssetPersistentAccess;
```

Comment:
Should this be renamed to `cpu_persistent_access` now that it doesnt return a bool?

Reply:
Eh I think unload_after_extract is a bit more clearer. If anything, I'd like to find a better name for RenderAssetPersistentAccess, but I don't want to block on it.

Reply:
may i suggest `RenderAssetPersistencePolicy` and `persistence_policy`?

Reply:
i mostly dont like that `unload_after_extract` sounds like a method with side effects, like you're telling it to unload, when really you're asking if it will

Reply:
Thanks for the suggestion, I agree it's better.